txpipe/maps.py

```diff
@@ -178,6 +162,29 @@ def prepare_mappers(self, pixel_scheme):
         )
         return [mapper, cal]
 
+    def get_calibrators(self):
+        shear_catalog_type = read_shear_catalog_type(self)
+        with self.open_input("shear_tomography_catalog") as f:
+            nbin_source = f["tomography"].attrs["nbin_source"]
+
+            if shear_catalog_type == "metacal":
+                R = f['/metacal_response/R_total'][:] # nbin x 2 x 2
+                cal = {i:R[i] for i in range(nbin_source)}
+                cal['2D'] = f['/metacal_response/R_total_2d'][:]
+            elif shear_catalog_type == "lensfit":
```

Comment:
Why does the lensfit option also have responses to calibrate? Isn't the difference between the first and second option to use the full matrix response or just the mean response?

Reply:
I don't know much about the lensfit cal - just adapting what @ellongley did but for the 2D case. There's a function in `calibration_tools.py` that handles it all